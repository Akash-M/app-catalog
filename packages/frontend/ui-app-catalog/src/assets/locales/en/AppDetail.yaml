header: App Detail
backButton: Back to App List
details:
  id: ID
  version: Version
  description: Description
  author: Author
  url: URL
  readme: ReadMe
