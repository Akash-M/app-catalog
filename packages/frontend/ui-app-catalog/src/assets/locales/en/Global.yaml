appName: App Catalog
routes:
  catalogList: App Catalog
