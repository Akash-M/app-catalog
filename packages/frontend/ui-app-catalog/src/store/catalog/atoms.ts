import { atom } from 'recoil';

export const AppListState = atom<AppCatalog.Catalog[]>({
  key: 'AppList',
  default: [],
});
