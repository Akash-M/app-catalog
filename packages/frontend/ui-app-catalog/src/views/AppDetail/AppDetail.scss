.app-detail {
  > button {
    align-items: center;
    border: none;
    background: none;
    color: $ac-blue;
    cursor: pointer;
    display: flex;
    flex-direction: row;
    font-size: pxToRem(16);
    font-weight: bold;
    padding: 0;

    > img {
      margin-right: pxToRem(8);
    }
  }

  > section {
    > img {
      margin-bottom: pxToRem(32);
      width: 50px;
    }

    > .info {
      background-color: $ac-grey-2;
      border-radius: pxToRem(16);
      margin-bottom: pxToRem(32);

      > div {
        align-items: flex-start;
        border-bottom: 1px solid $ac-black-2;
        display: flex;
        flex-direction: column;
        padding: 0 1rem;

        @include tablet {
          align-items: center;
          flex-direction: row;
        }

        > h4 {
          color: $ac-black-3;
          margin-bottom: pxToRem(8);
          width: 100px;

          @include tablet {
            margin: pxToRem(16);
          }
        }

        &:last-child {
          border-bottom: none;
        }

        > p, a {
          margin: 0 0 pxToRem(8) 0;

          @include tablet {
            margin: pxToRem(16);
          }
        }
      }
    }
  }

  > .readme {
    background-color: $ac-grey-2;
    border-radius: pxToRem(16);
    cursor: not-allowed;
    padding: 1rem;
    pointer-events: none;
  }
}
