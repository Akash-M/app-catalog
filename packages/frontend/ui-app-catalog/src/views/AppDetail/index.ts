export { AppDetail as default } from '$/views/AppDetail/AppDetail';
