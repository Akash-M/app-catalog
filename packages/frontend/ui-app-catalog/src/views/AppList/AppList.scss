.app-list {
  &__tiles {
    display: flex;
    flex-flow: row wrap;

    > .ac-tile {
      margin: pxToRem(16);
    }
  }
}
