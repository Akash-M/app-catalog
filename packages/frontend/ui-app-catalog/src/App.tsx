import React, { useEffect } from 'react';
import { ToastContainer } from 'react-toastify';
import 'react-toastify/dist/ReactToastify.css';
import { useSetRecoilState } from 'recoil';

import { getApps } from 'lib-api/src/catalog';

import '$/App.scss';
import Layout from '$/containers/Layout';
import { AppListState } from '$/store/catalog/atoms';

const App = (): JSX.Element => {
  const setCatalogList = useSetRecoilState(AppListState);

  const fetchCatalogList = async () => {
    setCatalogList(await getApps());
  };

  useEffect(() => {
    void fetchCatalogList();
  }, []);

  return (
    <main>
      <Layout />

      <ToastContainer />
    </main>
  );
};

export default App;
