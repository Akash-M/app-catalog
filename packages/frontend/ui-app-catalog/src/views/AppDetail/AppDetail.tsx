import React, { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import ReactMarkdown from 'react-markdown';
import { useParams } from 'react-router-dom';

import { getAppById, getAppReadme } from 'lib-api/src/catalog';

import './AppDetail.scss';

export function AppDetail(): JSX.Element {
  const { t } = useTranslation(['Global', 'AppDetail']);
  // TODO: fix type.
  const { id } = useParams<any>();
  const [appDetail, setAppDetail] = useState<AppCatalog.Catalog>();
  const [appReadme, setAppReadme] = useState<string>('');
  // TODO: specify error types.
  const [error, setError] = useState(false);

  const fetchAppDetail = async () => {
    try {
      if (id) {
        const response = await getAppById(id);
        setAppDetail(response);
        if (response.readmeURL) {
          setAppReadme(await getAppReadme(response.readmeURL));
        }
      } else {
        setError(true);
      }
    } catch {
      setError(true);
    }
  };

  useEffect(() => {
    void fetchAppDetail();
  }, [id]);

  return (
    <article className="app-detail">
      <h2>{t('AppDetail.header')}</h2>

      {appDetail && !error && (
        <section className="app-detail__info">
          {appDetail.iconURL && (
            <img
              alt={appDetail.iconURL}
              className="icon"
              src={appDetail.iconURL}
            />
          )}

          <p className="title">{appDetail.name}</p>

          <p>ID: {appDetail.id}</p>

          <p>Version: {appDetail.version}</p>

          <p>Description: {appDetail.description}</p>

          {appDetail.author && <p>Author: {appDetail.author}</p>}

          {appDetail.url && (
            <p>
              <a href={appDetail.url} rel="noreferrer" target="_blank">
                {appDetail.url}
              </a>
            </p>
          )}

          {appDetail.readmeURL && <ReactMarkdown>{appReadme}</ReactMarkdown>}
        </section>
      )}
    </article>
  );
}
