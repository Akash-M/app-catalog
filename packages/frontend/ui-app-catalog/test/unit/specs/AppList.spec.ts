import { fireEvent, waitFor } from '@testing-library/react';
import { setI18n } from 'react-i18next';
import selectEvent from 'react-select-event';

import { getAppById, getAppReadme, getApps } from 'lib-api/src/catalog';
import { customRenderer } from 'lib-utils/src/testing/factory';
import { I18N_MISSING_KEY, loadI18n } from 'lib-utils/src/testing/i18n';

import App from '$/App';
import AppDetail from '$/assets/locales/en/AppDetail.yaml';
import AppList from '$/assets/locales/en/AppList.yaml';
import Global from '$/assets/locales/en/Global.yaml';
import { AppListState } from '$/store/catalog/atoms';
import { appListFixtures } from '#/unit/fixtures/app-list';

jest.mock('lib-api/src/catalog', () => ({
  getAppById: jest.fn(),
  getAppReadme: jest.fn(),
  getApps: jest.fn(),
}));

const initializeState = ({ set }: any) => {
  set(AppListState, appListFixtures);
};

describe('<App />', () => {
  beforeAll(() => {
    setI18n(loadI18n('Global', { Global, AppDetail, AppList }));
  });

  beforeEach(jest.clearAllMocks);

  test('should render list of apps on load', async () => {
    (getApps as jest.Mock).mockResolvedValueOnce(appListFixtures);
    const { container } = customRenderer(App, initializeState);
    await waitFor(() => expect(getApps as jest.Mock).toHaveBeenCalledTimes(1));
    expect(container.firstChild!.textContent).not.toContain(I18N_MISSING_KEY);
    expect(container.firstChild).toMatchSnapshot();
    expect(container.querySelectorAll('.ac-tile').length).toBe(8);
  });

  test('should render app list filtered by author', async () => {
    (getApps as jest.Mock).mockResolvedValueOnce(appListFixtures);
    const { container, getByText } = customRenderer(App, initializeState);
    await waitFor(() => expect(getApps as jest.Mock).toHaveBeenCalledTimes(1));
    const authorFilter = await waitFor(() => getByText('Filter by Author'));
    await selectEvent.select(authorFilter, 'Giantswarm');
    expect(container.querySelectorAll('.ac-tile').length).toBe(4);
    await selectEvent.clearAll(getByText('Giantswarm'));
    expect(container.querySelectorAll('.ac-tile').length).toBe(8);
  });

  test('should display filtered app list in global search', async () => {
    (getAppById as jest.Mock).mockResolvedValueOnce(appListFixtures[0]);
    (getApps as jest.Mock).mockResolvedValueOnce(appListFixtures);
    (getAppReadme as jest.Mock).mockResolvedValueOnce('mock-read-me');
    const { container, getByText } = customRenderer(App, initializeState);
    await waitFor(() => expect(getApps as jest.Mock).toHaveBeenCalledTimes(1));
    const node = await waitFor(() => getByText('Search'));
    await selectEvent.select(node, 'prometheus-operator-app-chart');
    fireEvent.click(getByText('prometheus-operator-app-chart'));
    await waitFor(() => {
      expect(getAppById as jest.Mock).toHaveBeenCalledTimes(1);
      expect(getAppReadme as jest.Mock).toHaveBeenCalledTimes(1);
    });
    expect(container.firstChild).toMatchSnapshot();
    await waitFor(() => {
      expect(getByText('AppDetail.backButton')).toBeTruthy();
    });
  });
});
