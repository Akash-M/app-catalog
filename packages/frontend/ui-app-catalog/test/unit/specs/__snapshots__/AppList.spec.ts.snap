// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should display filtered app list in global search 1`] = `
<main>
  <article
    class="layout"
  >
    <nav
      class="site-header"
    >
      <a
        class=""
        href="/"
      >
        <h2>
          Global.appName
        </h2>
      </a>
      <div
        class="site-header__search css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-6-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class=" css-1s2u09g-control"
        >
          <div
            class=" css-319lph-ValueContainer"
          >
            <div
              class=" css-qc6sy-singleValue"
            >
              prometheus-operator-app-chart
            </div>
            <div
              class=" css-6j8wv5-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-controls="react-select-6-listbox"
                aria-expanded="false"
                aria-haspopup="true"
                aria-owns="react-select-6-listbox"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class=""
                id="react-select-6-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 0; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <section
      class="layout__screens"
    >
      <article
        class="app-detail"
      >
        <button>
          <img
            alt="left-arrow"
            src="assets/images/arrow-left.svg"
          />
          AppDetail.backButton
        </button>
        <section>
          <h2>
            prometheus-operator-app-chart
          </h2>
          <img
            alt="https://raw.githubusercontent.com/prometheus/prometheus.github.io/master/assets/prometheus_logo-cb55bb5c346.png"
            class="icon"
            src="https://raw.githubusercontent.com/prometheus/prometheus.github.io/master/assets/prometheus_logo-cb55bb5c346.png"
          />
          <div
            class="info"
          >
            <div>
              <h4>
                AppDetail.details.id
              </h4>
               
              <p>
                asdm123312asdnboqwedsa
              </p>
            </div>
            <div>
              <h4>
                AppDetail.details.version
              </h4>
               
              <p>
                0.2.0
              </p>
            </div>
            <div>
              <h4>
                AppDetail.details.description
              </h4>
               
              <p>
                A Helm Chart for Prometheus Operator.
              </p>
            </div>
            <div>
              <h4>
                AppDetail.details.author
              </h4>
               
              <p>
                Giantswarm
              </p>
            </div>
            <div>
              <h4>
                AppDetail.details.url
              </h4>
              <a
                href="https://github.com/giantswarm/prometheus-operator-app"
                rel="noreferrer"
                target="_blank"
              >
                https://github.com/giantswarm/prometheus-operator-app
              </a>
            </div>
          </div>
        </section>
        <section
          class="readme"
        >
          <h4>
            AppDetail.details.readme
          </h4>
          <p>
            mock-read-me
          </p>
        </section>
      </article>
    </section>
  </article>
  <div
    class="Toastify"
  />
</main>
`;

exports[`<App /> should render list of apps on load 1`] = `
<main>
  <article
    class="layout"
  >
    <nav
      class="site-header"
    >
      <a
        aria-current="page"
        class="active"
        href="/"
      >
        <h2>
          Global.appName
        </h2>
      </a>
      <div
        class="site-header__search css-b62m3t-container"
      >
        <span
          class="css-1f43avz-a11yText-A11yText"
          id="react-select-2-live-region"
        />
        <span
          aria-atomic="false"
          aria-live="polite"
          aria-relevant="additions text"
          class="css-1f43avz-a11yText-A11yText"
        />
        <div
          class=" css-1s2u09g-control"
        >
          <div
            class=" css-319lph-ValueContainer"
          >
            <div
              class=" css-14el2xx-placeholder"
              id="react-select-2-placeholder"
            >
              Search
            </div>
            <div
              class=" css-6j8wv5-Input"
              data-value=""
            >
              <input
                aria-autocomplete="list"
                aria-controls="react-select-2-listbox"
                aria-describedby="react-select-2-placeholder"
                aria-expanded="false"
                aria-haspopup="true"
                aria-owns="react-select-2-listbox"
                autocapitalize="none"
                autocomplete="off"
                autocorrect="off"
                class=""
                id="react-select-2-input"
                role="combobox"
                spellcheck="false"
                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                tabindex="0"
                type="text"
                value=""
              />
            </div>
          </div>
          <div
            class=" css-1hb7zxy-IndicatorsContainer"
          >
            <span
              class=" css-1okebmr-indicatorSeparator"
            />
            <div
              aria-hidden="true"
              class=" css-tlfecz-indicatorContainer"
            >
              <svg
                aria-hidden="true"
                class="css-tj5bde-Svg"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <section
      class="layout__screens"
    >
      <article
        class="app-list"
      >
        <h2>
          AppList.header
        </h2>
        <section
          class="app-list__filters"
        >
          <div
            class="filter css-b62m3t-container"
          >
            <span
              class="css-1f43avz-a11yText-A11yText"
              id="react-select-3-live-region"
            />
            <span
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
              class="css-1f43avz-a11yText-A11yText"
            />
            <div
              class=" css-1s2u09g-control"
            >
              <div
                class=" css-319lph-ValueContainer"
              >
                <div
                  class=" css-14el2xx-placeholder"
                  id="react-select-3-placeholder"
                >
                  Filter by Author
                </div>
                <div
                  class=" css-6j8wv5-Input"
                  data-value=""
                >
                  <input
                    aria-autocomplete="list"
                    aria-controls="react-select-3-listbox"
                    aria-describedby="react-select-3-placeholder"
                    aria-expanded="false"
                    aria-haspopup="true"
                    aria-owns="react-select-3-listbox"
                    autocapitalize="none"
                    autocomplete="off"
                    autocorrect="off"
                    class=""
                    id="react-select-3-input"
                    role="combobox"
                    spellcheck="false"
                    style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class=" css-1hb7zxy-IndicatorsContainer"
              >
                <span
                  class=" css-1okebmr-indicatorSeparator"
                />
                <div
                  aria-hidden="true"
                  class=" css-tlfecz-indicatorContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="css-tj5bde-Svg"
                    focusable="false"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                  >
                    <path
                      d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                    />
                  </svg>
                </div>
              </div>
            </div>
            <div>
              <input
                name="author-filter"
                type="hidden"
              />
            </div>
          </div>
        </section>
        <section
          class="app-list__tiles"
        >
          <div
            class="ac-tile"
          >
            <div
              class="ac-tile__header"
            >
              <img
                alt="https://raw.githubusercontent.com/prometheus/prometheus.github.io/master/assets/prometheus_logo-cb55bb5c346.png"
                class="icon"
                src="https://raw.githubusercontent.com/prometheus/prometheus.github.io/master/assets/prometheus_logo-cb55bb5c346.png"
              />
              <p
                class="title"
              >
                prometheus-operator-app-chart
              </p>
            </div>
            <div
              class="ac-tile__details"
            >
              <p>
                Description: 
                A Helm Chart for Prometheus Operator.
              </p>
              <p>
                Author: 
                Giantswarm
              </p>
            </div>
            <div
              class="ac-tile__chips"
            >
              <span
                class="ac-chip ac-chip--available"
              >
                URL available
              </span>
              <span
                class="ac-chip ac-chip--available"
              >
                ReadMe Available
              </span>
            </div>
          </div>
          <div
            class="ac-tile"
          >
            <div
              class="ac-tile__header"
            >
              <img
                alt="https://upload.wikimedia.org/wikipedia/commons/thumb/c/c5/Nginx_logo.svg/500px-Nginx_logo.svg.png"
                class="icon"
                src="https://upload.wikimedia.org/wikipedia/commons/thumb/c/c5/Nginx_logo.svg/500px-Nginx_logo.svg.png"
              />
              <p
                class="title"
              >
                nginx-ingress-controller-app
              </p>
            </div>
            <div
              class="ac-tile__details"
            >
              <p>
                Description: 
                A Helm chart for the nginx ingress-controller.
              </p>
              <p>
                Author: 
                Giantswarm
              </p>
            </div>
            <div
              class="ac-tile__chips"
            >
              <span
                class="ac-chip ac-chip--available"
              >
                URL available
              </span>
              <span
                class="ac-chip ac-chip--available"
              >
                ReadMe Available
              </span>
            </div>
          </div>
          <div
            class="ac-tile"
          >
            <div
              class="ac-tile__header"
            >
              <img
                alt="https://s3.amazonaws.com/downloads.kong/universe/assets/icon-kong-inc-large.png"
                class="icon"
                src="https://s3.amazonaws.com/downloads.kong/universe/assets/icon-kong-inc-large.png"
              />
              <p
                class="title"
              >
                kong-app
              </p>
            </div>
            <div
              class="ac-tile__details"
            >
              <p>
                Description: 
                The Cloud-Native Ingress and Service Mesh for APIs and Microservices.
              </p>
              <p>
                Author: 
                Kong
              </p>
            </div>
            <div
              class="ac-tile__chips"
            >
              <span
                class="ac-chip ac-chip--available"
              >
                URL available
              </span>
              <span
                class="ac-chip ac-chip--unavailable"
              >
                ReadMe Unavailable
              </span>
            </div>
          </div>
          <div
            class="ac-tile"
          >
            <div
              class="ac-tile__header"
            >
              <img
                alt="https://github.com/giantswarm/efk-stack-app/raw/master/images/logo.png"
                class="icon"
                src="https://github.com/giantswarm/efk-stack-app/raw/master/images/logo.png"
              />
              <p
                class="title"
              >
                efk-stack-app
              </p>
            </div>
            <div
              class="ac-tile__details"
            >
              <p>
                Description: 
                Open Distro for Elasticsearch.
              </p>
              <p>
                Author: 
                Giantswarm
              </p>
            </div>
            <div
              class="ac-tile__chips"
            >
              <span
                class="ac-chip ac-chip--available"
              >
                URL available
              </span>
              <span
                class="ac-chip ac-chip--available"
              >
                ReadMe Available
              </span>
            </div>
          </div>
          <div
            class="ac-tile"
          >
            <div
              class="ac-tile__header"
            >
              <img
                alt="https://raw.githubusercontent.com/jetstack/cert-manager/master/logo/logo.png"
                class="icon"
                src="https://raw.githubusercontent.com/jetstack/cert-manager/master/logo/logo.png"
              />
              <p
                class="title"
              >
                cert-manager-app
              </p>
            </div>
            <div
              class="ac-tile__details"
            >
              <p>
                Description: 
                A Helm chart for cert-manager.
              </p>
              <p>
                Author: 
                Giantswarm
              </p>
            </div>
            <div
              class="ac-tile__chips"
            >
              <span
                class="ac-chip ac-chip--available"
              >
                URL available
              </span>
              <span
                class="ac-chip ac-chip--available"
              >
                ReadMe Available
              </span>
            </div>
          </div>
          <div
            class="ac-tile"
          >
            <div
              class="ac-tile__header"
            >
              <img
                alt="https://www.aquasec.com/wp-content/uploads/2016/05/aqua_logo_fullcolor.png"
                class="icon"
                src="https://www.aquasec.com/wp-content/uploads/2016/05/aqua_logo_fullcolor.png"
              />
              <p
                class="title"
              >
                aqua-app-server
              </p>
            </div>
            <div
              class="ac-tile__details"
            >
              <p>
                Description: 
                A Helm chart for the Aqua Console, Gateway Components.
              </p>
              <p>
                Author: 
                Aqua
              </p>
            </div>
            <div
              class="ac-tile__chips"
            >
              <span
                class="ac-chip ac-chip--available"
              >
                URL available
              </span>
              <span
                class="ac-chip ac-chip--unavailable"
              >
                ReadMe Unavailable
              </span>
            </div>
          </div>
          <div
            class="ac-tile"
          >
            <div
              class="ac-tile__header"
            >
              <img
                alt="https://www.aquasec.com/wp-content/uploads/2016/05/aqua_logo_fullcolor.png"
                class="icon"
                src="https://www.aquasec.com/wp-content/uploads/2016/05/aqua_logo_fullcolor.png"
              />
              <p
                class="title"
              >
                aqua-app-scanner
              </p>
            </div>
            <div
              class="ac-tile__details"
            >
              <p>
                Description: 
                A Helm chart for the Aqua Scanner Components.
              </p>
              <p>
                Author: 
                Aqua
              </p>
            </div>
            <div
              class="ac-tile__chips"
            >
              <span
                class="ac-chip ac-chip--available"
              >
                URL available
              </span>
              <span
                class="ac-chip ac-chip--unavailable"
              >
                ReadMe Unavailable
              </span>
            </div>
          </div>
          <div
            class="ac-tile"
          >
            <div
              class="ac-tile__header"
            >
              <img
                alt="https://www.aquasec.com/wp-content/uploads/2016/05/aqua_logo_fullcolor.png"
                class="icon"
                src="https://www.aquasec.com/wp-content/uploads/2016/05/aqua_logo_fullcolor.png"
              />
              <p
                class="title"
              >
                aqua-app-enforcer
              </p>
            </div>
            <div
              class="ac-tile__details"
            >
              <p>
                Description: 
                A Helm chart for the Aqua Enforcer Components.
              </p>
              <p>
                Author: 
                Aqua
              </p>
            </div>
            <div
              class="ac-tile__chips"
            >
              <span
                class="ac-chip ac-chip--available"
              >
                URL available
              </span>
              <span
                class="ac-chip ac-chip--unavailable"
              >
                ReadMe Unavailable
              </span>
            </div>
          </div>
        </section>
      </article>
    </section>
  </article>
  <div
    class="Toastify"
  />
</main>
`;
